Source/p_maputl.c

```diff
@@ -386,7 +398,8 @@ boolean P_BlockLinesIterator(int x, int y, boolean func(line_t*))
   // Most demos go out of sync, and maybe other problems happen, if we
   // don't consider linedef 0. For safety this should be qualified.
 
-  if (!demo_compatibility) // killough 2/22/98: demo_compatibility check
+  // mbf21: Fix blockmap issue seen in btsx e2 Map 20
+  if (!demo_compatibility && !mbf21) // killough 2/22/98: demo_compatibility check
```

Comment:
I believe there is/was another check that at least one block does not have `0` as first element in the list?

Reply:
Yes, in the P_VerifyBlockMap function, which we don't have. Not sure if this is necessary.

Reply:
> Yes, in the P_VerifyBlockMap function, which we don't have. Not sure if this is necessary.

I believe it is important to check if the first element of each list in the blockmap is always `0` or not - not matter if we have the function in the code where @kraflab implemented this check or not. Ryan, what do you say?

Reply:
I think this fix is from Eternity and the P_VerifyBlockMap function is from there too.

Reply:
I think you might still need it? The default for mbf21 is to skip the first element, like vanilla (the opposite of what is currently here). If P_VerifyBlockMap finds a nonzero first element then it overrides that behaviour and doesn't skip it (the mbf default).